domain/ide/workspace/ui/diagnostics_overview.go

```diff
@@ -0,0 +1,204 @@
+/*
+ * © 2024 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ui
+
+import (
+	"bytes"
+	"crypto/rand"
+	_ "embed"
+	"encoding/base64"
+	"fmt"
+	"html/template"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/domain/snyk"
+	"github.com/snyk/snyk-ls/internal/html"
+	"github.com/snyk/snyk-ls/internal/notification"
+	"github.com/snyk/snyk-ls/internal/product"
+	"github.com/snyk/snyk-ls/internal/types"
+)
+
+//go:embed template/diagnosticsOverview.html
+var diagnosticsOverviewTemplatePath string
+
+//go:embed template/diagnosticsOverview.css
+var diagnosticsOverviewTemplateCSS string
+var diagnosticsOverviewTemplate *template.Template
+
+func init() {
+	logger := config.CurrentConfig().Logger().With().Str("method", "ui.init").Logger()
```

Comment:
inject the config instead of singleton?

Reply:
Can't change the signature of `init()`, else I'd be with you :)

Reply:
We could move the template parsing into a singleton though, if it's important.